agent/utils/ticker_test.go

```diff
@@ -0,0 +1,66 @@
+package utils
+
+import (
+	"context"
+	"math/rand"
+	"testing"
+	"time"
+)
+
+const (
+	TheBestNumber = 28
+)
+
+func TestTickerHappyCase(t *testing.T) {
+	ctx, _ := context.WithTimeout(context.Background(), 1000 * time.Millisecond)
+	mTicker := JitteredTicker(ctx, 10*time.Millisecond, 100*time.Millisecond)
+
+	times := 0
+	for {
+		_, ok := <-mTicker
+		times++
+		if !ok {
+			break
+		}
+	}
+
+	if 10 > times || times > 100  {
+		t.Error("Should tick at least 10 but less than 100 times: ", times)
+	}
+}
+
+func TestRandomDuration(t *testing.T) {
+	// The best randomness is deterministic
+	rand.Seed(TheBestNumber)
+
+	start := 10 * time.Second
+	end := 20 * time.Second
+
+	for i := 0; i < 10000; i++ {
+		c := randomDuration(start, end)
+		if c < start || end < c {
+			t.Errorf("Generated a bad time! seconds: %v nanoseconds: %d", c, c.Nanoseconds())
+		}
+	}
+}
+
+func TestSendNowOnlyPublishesOneEvent(t *testing.T) {
```

Comment:
I don't think the name is correct, if there are multiple goroutine listening to the channel, it could publish to each of these goroutines. If I understand correctly, you are testing the SendNow is not blocked even there is no listener.

Reply:
Yes!!! I'll fix that